rust/src/collection.rs

```diff
@@ -78,3 +91,76 @@ pub struct Label {
     pub id: i64,
     pub resource_url: String,
 }
+
+fn clean_artist_name(artist: String) -> String {
+    let regex = Regex::new(r"\s*\(\d+\)\s*").unwrap();
+    let result = regex.replace_all(artist.as_str(), "");
+        return result.to_string();
+}
+
+fn format_release(release: Album) -> String {
+    let title: String = release.basic_information.title;
+    let artist: String = release
+        .basic_information
+        .artists
+        .iter()
+        .map(|artist: &Artist| clean_artist_name(artist.name.clone()))
+        .collect::<Vec<_>>()
+        .join(" & ");
+
+    format!("{} - {}", artist, title)
+}
+
+fn get_random_releases(releases: Vec<Album>) -> Vec<Album> {
+    let mut random_releases: Vec<Album> = Vec::new();
+    for _ in 0..4 {
+        let random_index = rand::thread_rng().gen_range(0..releases.len());
+        random_releases.push(releases[random_index].clone());
+    }
```

Comment:
The odds of this happening are pretty low, but if you want to ensure four distinct random indices (without repeats) you can use `choose_multiple`: https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.choose_multiple

https://chatgpt.com/share/6702eb23-4d58-8006-869d-06e829cbb4b4

Reply:
done